vtm/src/org/oscim/layers/tile/TileManager.java

```diff
@@ -56,6 +60,12 @@
     private int mMinZoom;
     private int mMaxZoom;
 
+    /**
+     * Internal sorted list, to save zoom limits, which some layers need to get rendered.
+     * Uses a <Code>List</Code> cause <Code>Set</Code> doesn't save doubles.
+     */
+    private List<Integer> mZoomLimits;
```

Comment:
If want an ordered collection with no duplicates, better use directly a [TreeSet](https://docs.oracle.com/javase/7/docs/api/java/util/TreeSet.html) instead (without later conversions).

Reply:
@devemux86: sorry I went through review too fast. This was misunderstood. I want to save duplicates! Reason is that there could have been multiple layers which add the same zoom limit, but if removing one of them, the other layer wouldn't get the zoom limits' tiles, even if zoom limit was added.
On the other hand this generally never happens, as layers aren't attached/removed while execution, so won't play a big role.

Reply:
Hmm valid argument, but in `update` a new `HashSet` is created each time. So need to think a bit the implementation.
Probably can maintain the set separately in add / remove to be available in all class, though doesn't seem nice to have 2 collections for same thing.

Reply:
Or can use a `HashMap` like done in [MultiMapFileTileSource](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/tiling/source/mapfile/MultiMapFileTileSource.java#L36) in a "similar" case.

Reply:
If it works, would be good (?). But `HashMap.values()` has duplicates too.

Or a sorted list and check while iterating if value equals the one before and skip.

But this seems the most efficient:
> Probably can maintain the Set separately in add / remove to be available in whole class, though doesn't seem nice to have 2 collections for same thing.